src/coreclr/jit/layout.cpp

```diff
@@ -778,6 +778,7 @@ ClassLayoutBuilder ClassLayoutBuilder::BuildArray(Compiler* compiler, CORINFO_CL
 
     ClrSafeInt<unsigned> totalSize(elementSize);
     totalSize *= static_cast<unsigned>(length);
+    totalSize.AlignUp(TARGET_POINTER_SIZE);
```

Comment:
@AndyAyersMS speaking of alignment - do you skip Aling8 types on 32bit? like array of doubles

Reply:
We handle those.

We don't model Align8 within the layout. It is a property of the symbol that has the layout. Though I suppose we could start having layouts with alignment demands...